packages/js/src/ai-generator/components/modal-content.js

```diff
@@ -14,51 +14,58 @@ export const ModalContent = () => {
 	const upsellLinkPremium = useSelect( select => select( STORE ).selectLink( "https://yoa.st/ai-generator-upsell" ), [] );
 	const upsellLinkWooPremiumBundle = useSelect( select => select( STORE ).selectLink( "https://yoa.st/ai-generator-upsell-woo-seo-premium-bundle" ), [] );
 	const upsellLinkWoo = useSelect( select => select( STORE ).selectLink( "https://yoa.st/ai-generator-upsell-woo-seo" ), [] );
-	const isPremium = useSelect( select => select( STORE ).getIsPremium(), [] );
-	const isWooSeoUpsell = useSelect( select => select( STORE ).getIsWooSeoUpsell(), [] );
-	const isProduct = useSelect( select => select( STORE ).getIsProduct(), [] );
-	const wooSeoNoPremium = isProduct && ! isWooSeoUpsell && ! isPremium;
-	const isProductCopy = !! ( isWooSeoUpsell || wooSeoNoPremium );
-	const postModalprops = {
-		isProductCopy,
+
+	const isPremiumActive = useSelect( select => select( STORE ).getIsPremium(), [] );
+	const isWooSeoActive = useSelect( select => select( STORE ).getIsWooSeoActive(), [] );
+	const isWooCommerceActive = useSelect( select => select( STORE ).getIsWooCommerceActive(), [] );
+
+	const isProductPost = useSelect( select => select( STORE ).getIsProduct(), [] );
+	const isProductTerm = useSelect( select => select( STORE ).getIsProductTerm(), [] );
+
+	const upsellProps = {
 		upsellLink: upsellLinkPremium,
 	};
 
-	if ( isProductCopy ) {
+	// Use specific copy for product posts.
+	if ( isWooCommerceActive && isProductPost ) {
+		upsellProps.title = __( "Generate product titles & descriptions with AI!", "wordpress-seo" );
+		upsellProps.isProductCopy = true;
```

Comment:
Just out of curiosity, trying to understand where the properties of the `upsellProps` object come from. Is it correct that you add/create them when needed and add a value to them? 🤔

Reply:
Yes, so `upsellProps` starts off as a JS object with just one attribute, and then we add further attributes as we go 😸 The object is then passed (destructured) to the `AiGenerateTitlesAndDescriptionsUpsell` and further processing is done there.